orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -40,11 +56,20 @@ def __repr__(self) -> str:
             \t#Positionals (non-None): {self.num_positionals_non_none}
             \tKeyword names:           {self.keyword_names}'''
 
-    def flatten(self, args, kwargs):
+    def flatten(self, args, kwargs, device):
         '''Flatten args and kwargs in a single tuple of tensors with strict ordering'''
 
-        ret = list(args)
+        ret = [None]*len(args)
+        for idx, arg in enumerate(args):
+            if _Primitive.is_primitive(arg):
+                ret[idx] = _Primitive.get_tensor(arg, device)
+            else:
+                ret[idx] = arg
```

Comment:
List comprehension performs better than regular loop with if-else. This perf gain is important as flattening happens for all steps